### ISSUE TITLE
[PATTERN]: Create an accessible Filter/Search pattern for inclusion in the design system

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/veteran-facing-services-tools#159]
**Is your feature request related to a problem? Please describe.**
The design team has identified several patterns for complex search queries and result set filters. During the 8/26/19 weekly call, the team determined it would be good to build an accessible filter component.

**Describe the solution you'd like**
* We would like to create an accessible component that can be configured to handle search queries and filter existing result sets.
* The component should be inclusive and pass all automated and manual 508 testing.

**Related Issues:**
* https://github.com/department-of-veterans-affairs/va.gov-team/issues/901 (blocked)
* https://github.com/department-of-veterans-affairs/va.gov-team/issues/1276 (blocked)
* https://github.com/department-of-veterans-affairs/va.gov-team/issues/731 (potentially related)

**Additional context**
* During the discussion, the AirBnB filter mechanism was highlighted as a good starting point. Screenshots attached below.

---

![Screen Shot 2019-08-26 at 1 39 59 PM](https://user-images.githubusercontent.com/934879/63714898-500f8a00-c808-11e9-803c-e9b678e451d6.png)

---
![Screen Shot 2019-08-26 at 1 39 51 PM](https://user-images.githubusercontent.com/934879/63714925-5998f200-c808-11e9-8564-8b316f30e11e.png)
